--- mrustc-1.54.0/rustc-1.54.0-src/library/unwind/src/lib.rs	2023-09-13 09:00:57.953000000 +1000
+++ mrustc-1.54.0/rustc-1.54.0-src/library/unwind/src/lib.rs	2023-09-13 17:54:36.544000000 +1000
@@ -49,7 +49,6 @@
         extern "C" {}
     } else {
-        #[link(name = "unwind", kind = "static", cfg(target_feature = "crt-static"))]
+        #[link(name = "unwind")]
-        #[link(name = "gcc_s", cfg(not(target_feature = "crt-static")))]
         extern "C" {}
     }
 }
@@ -72,8 +71,7 @@
     not(feature = "llvm-libunwind"),
     feature = "system-llvm-libunwind"
 ))]
-#[link(name = "unwind", cfg(not(target_feature = "crt-static")))]
-extern "C" {}
+compile_error!("target_env is gnu");
 
 #[cfg(target_os = "redox")]
 #[link(name = "gcc_eh", kind = "static-nobundle", cfg(target_feature = "crt-static"))]
