--- a/test/Makefile.am	2026-02-09 20:11:10.625581204 +1000
+++ b/test/Makefile.am	2026-02-09 20:19:50.646546845 +1000
@@ -1563,7 +1563,7 @@
 	forsimp fsbs fscaret fsnul1 fsrs fsspcoln fstabplus funsemnl \
 	funsmnam funstack \
 	getline getline2 getline3 getline4 getline5 getlnbuf getlnfa \
-	getnr2tb getnr2tm gsubasgn gsubnulli18n gsubtest gsubtst2 gsubtst3 \
+	getnr2tb getnr2tm gsubasgn gsubtest gsubtst2 gsubtst3 \
 	gsubtst4 gsubtst5 gsubtst6 gsubtst7 gsubtst8 \
 	hex hex2 hsprint \
 	inpref inputred intest intprec iobug1 \
@@ -1605,7 +1605,7 @@
 	argtest arraysort arraysort2 arraytype asortbool asortsymtab \
 	backw badargs beginfile1 beginfile2 binmode1 \
 	charasbytes clos1way clos1way2 clos1way3 clos1way4 clos1way5 \
-	clos1way6 colonwarn commas crlf csv1 csv2 csv3 csvodd \
+	clos1way6 colonwarn crlf csv1 csv2 csv3 csvodd \
 	dbugarray1 dbugarray2 dbugarray3 dbugarray4 dbugeval dbugeval2 \
 	dbugeval3 dbugeval4 dbugtypedre1 dbugtypedre2 delmessy delsub \
 	devfd devfd1 devfd2 dfacheck1 dumpvars elemnew1 \
@@ -1621,7 +1621,7 @@
 	indirectcall3 intarray iolint isarrayunset \
 	lint lintexp lintindex lintint lintlength lintold lintplus \
 	lintplus2 lintplus3 lintset lintwarn manyfiles \
-	match1 match2 match3 mbstr1 mbstr2 mdim1 mdim2 mdim3 mdim4 mdim5 \
+	match1 match2 match3 mdim1 mdim2 mdim3 mdim4 mdim5 \
 	mdim6 mdim7 mdim8 memleak2 memleak3 mixed1 mktime modifiers \
 	muldimposix \
 	nastyparm negtime next nondec nondec2 nonfatal1 nonfatal2 \
@@ -1644,7 +1644,6 @@
 	timeout typedregex1 typedregex2 typedregex3 typedregex4 \
 	typedregex5 typedregex6 typeof1 typeof2 typeof3 typeof4 typeof5 \
 	typeof6 typeof7 typeof8 typeof9 \
-	unicode1 \
 	watchpoint1
 
 ARRAYDEBUG_TESTS = arrdbg

