Patch origin https://github.com/pallets/click/pull/3238
diff --git a/tests/test_utils.py b/tests/test_utils.py
index e941415..1186ed3 100644
--- a/tests/test_utils.py
+++ b/tests/test_utils.py
@@ -7,8 +7,6 @@ from collections import namedtuple
 from contextlib import nullcontext
 from functools import partial
 from io import StringIO
-from pathlib import Path
-from tempfile import tempdir
 from unittest.mock import patch
 
 import pytest
@@ -328,7 +326,7 @@ EchoViaPagerTest = namedtuple(
         ),
     ],
 )
-def test_echo_via_pager(monkeypatch, capfd, pager_cmd, test):
+def test_echo_via_pager(monkeypatch, capfd, pager_cmd, test, tmp_path):
     monkeypatch.setitem(os.environ, "PAGER", pager_cmd)
     monkeypatch.setattr(click._termui_impl, "isatty", lambda x: True)
 
@@ -340,8 +338,7 @@ def test_echo_via_pager(monkeypatch, capfd, pager_cmd, test):
 
     check_raise = pytest.raises(expected_error) if expected_error else nullcontext()
 
-    pager_out_tmp = Path(tempdir) / "pager_out.txt"
-    pager_out_tmp.unlink(missing_ok=True)
+    pager_out_tmp = tmp_path / "pager_out.txt"
     with pager_out_tmp.open("w") as f:
         force_subprocess_stdout = patch.object(
             subprocess,
