diff --git a/meson.build b/meson.build
index c098c11..989d55c 100644
--- a/meson.build
+++ b/meson.build
@@ -21,11 +21,11 @@ lua_dep = dependency(lua_pc_name != '' ? lua_pc_name : lua_name, required: get_o
 
 python_dep = dependency('python3', required: get_option('python'))
 scdoc_dep = dependency('scdoc', version: '>=1.10', required: get_option('docs'), native: true)
-zlib_dep = dependency('zlib')
-libzstd_dep = dependency('libzstd', required: get_option('zstd'))
+zlib_dep = dependency('zlib', static: get_option('static'))
+libzstd_dep = dependency('libzstd', required: get_option('zstd'), static: get_option('static'))
 
 if get_option('crypto_backend') == 'openssl'
-	crypto_dep = dependency('openssl')
+	crypto_dep = dependency('openssl', static: get_option('static'))
 elif get_option('crypto_backend') == 'mbedtls'
 	crypto_dep = [ dependency('mbedtls'), dependency('mbedcrypto') ]
 endif
diff --git a/meson_options.txt b/meson_options.txt
index 6341145..854f597 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -14,3 +14,4 @@ option('tests', description: 'Build tests', type: 'feature', value: 'auto')
 option('url_backend', description: 'URL backend', type: 'combo', choices: ['libfetch', 'wget'], value: 'libfetch')
 option('uvol_db_target', description: 'Default target for uvol database layer', type: 'string')
 option('zstd', description: 'Build with zstd support', type: 'feature', value: 'auto', deprecated: {'true': 'enabled', 'false': 'disabled'})
+option('static', description: 'Link system libraries statically', type: 'boolean', value: false)
