From 9e8a09cc1862cc7c98460820860b2481d5def60f Mon Sep 17 00:00:00 2001
From: fundawang <fundawang@yeah.net>
Date: Sun, 13 Jul 2025 08:59:59 +0000
Subject: [PATCH] tests: only do sanity-lib test when tpm-tss is dlopened

Signed-off-by: Funda Wang <fundawang@yeah.net>
---
 tests/Makefile.am | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/tests/Makefile.am b/tests/Makefile.am
index b350977b95..1e8df5c6b6 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -552,7 +552,15 @@ dist_check_SCRIPTS += fastopen.sh pkgconfig.sh starttls.sh starttls-ftp.sh start
 	sni-resume.sh ocsp-tests/ocsptool.sh cert-reencoding.sh pkcs7-cat.sh long-crl.sh \
 	serv-udp.sh logfile-option.sh gnutls-cli-resume.sh profile-tests.sh \
 	server-weak-keys.sh ocsp-tests/ocsp-signer-verify.sh cfg-test.sh \
-	sanity-lib.sh gnutls-cli-earlydata.sh
+	gnutls-cli-earlydata.sh
+
+if ENABLE_TPM2
+if TPM2_ENABLE_DLOPEN
+dist_check_SCRIPTS += sanity-lib.sh
+endif
+else
+dist_check_SCRIPTS += sanity-lib.sh
+endif
 
 if !DISABLE_SYSTEM_CONFIG
 dist_check_SCRIPTS += system-override-sig.sh system-override-hash.sh \
-- 
GitLab

