diff --git a/third_party/devtools-frontend/src/third_party/typescript/typescript.py b/third_party/devtools-frontend/src/third_party/typescript/typescript.py
index b4bade9..ab31589 100644
--- a/third_party/devtools-frontend/src/third_party/typescript/typescript.py
+++ b/third_party/devtools-frontend/src/third_party/typescript/typescript.py
@@ -10,25 +10,7 @@ import os
 
 
 def GetBinaryPath():
-    if platform.machine() == 'arm64':
-        darwin_path = 'mac-arm64'
-    else:
-        darwin_path = 'mac-amd64'
-
-    relative = {
-        'Darwin': (darwin_path, 'src', 'lib', 'tsc'),
-        'Linux': ('linux-amd64', 'src', 'lib', 'tsc'),
-        'Windows': ('windows-amd64', 'src', 'lib', 'tsc.exe'),
-    }[platform.system()]
-
-    devtools_checkout_path = os_path.join(os_path.dirname(__file__), *relative)
-    if os_path.exists(devtools_checkout_path):
-        return os_path.normpath(devtools_checkout_path)
-
-    # Assume we are in a chromium checkout.
-    return os_path.normpath(
-        os_path.join(os_path.dirname(__file__), '..', '..', '..', '..',
-                     'typescript', *relative))
+    return "/usr/bin/tsc"
 
 
 def RunTypeScriptRaw(cmd_parts, stdout=None):
diff --git a/third_party/typescript/typescript.py b/third_party/typescript/typescript.py
index 7c2db71..962580f 100755
--- a/third_party/typescript/typescript.py
+++ b/third_party/typescript/typescript.py
@@ -11,16 +11,7 @@ import os
 
 
 def GetBinaryPath():
-  if platform.machine() == 'arm64':
-    darwin_path = 'mac-arm64'
-  else:
-    darwin_path = 'mac-amd64'
-
-  return os_path.normpath(os_path.join(os_path.dirname(__file__), *{
-    'Darwin': (darwin_path, 'src', 'lib', 'tsc'),
-    'Linux': ('linux-amd64', 'src', 'lib', 'tsc'),
-    'Windows': ('windows-amd64', 'src', 'lib', 'tsc.exe'),
-  }[platform.system()]))
+  return "/usr/bin/tsc"
 
 
 def RunTypeScriptRaw(cmd_parts, stdout=None):
