--- a/build/toolchain/toolchain.gni	2026-08-31 12:41:05.804703154 +1000
+++ b/build/toolchain/toolchain.gni	2026-08-31 12:41:25.443572929 +1000
@@ -41,12 +41,7 @@
 }
 
 declare_args() {
-  if (llvm_force_head_revision) {
-    clang_version = "24"
-  } else {
-    # TODO(crbug.com/534655507): Remove in the next Clang roll.
-    clang_version = "23"
-  }
+  clang_version = "22"
 }
 
 # Extension for shared library files (including leading dot).
