diff --git a/config.h b/config.h
index 62cc0ff..e26c0ce 100644
--- a/config.h
+++ b/config.h
@@ -1,5 +1,3 @@
-/* customized for musl setups, tweak for others */
-
 /* pick one */
 #define HAVE_CURSES_H 1
 /* #undef HAVE_NCURSES_H */
@@ -12,7 +10,7 @@
 #define HAVE_TERMCAP_H 1
 #define HAVE_TERM_H 1
 
-#define HAVE_ISSETUGID 1
+#undef HAVE_ISSETUGID
 /* #undef HAVE_STRUCT_DIRENT_D_NAMLEN */
 
 #ifndef _GNU_SOURCE

