commit b43139c14f3344dd29c50e77a3d135a2c21c8f84
Author: Wesley Moore <wes@wezm.net>
Date:   Mon Mar 2 17:59:48 2026 +1000

    Just make everything use POSIX getopt on glibc

diff --git a/src.freebsd/apply/apply.c b/src.freebsd/apply/apply.c
index 1b98e7a..81e8618 100644
--- a/src.freebsd/apply/apply.c
+++ b/src.freebsd/apply/apply.c
@@ -140,7 +140,7 @@ main(int argc, char *argv[])
 	debug = 0;
 	magic = '%';		/* Default magic char is `%'. */
 	nargs = -1;
-	while ((ch = getopt(argc, argv, "a:d0123456789")) != -1)
+	while ((ch = getopt(argc, argv, "+a:d0123456789")) != -1)
 		switch (ch) {
 		case 'a':
 			if (optarg[0] == '\0' || optarg[1] != '\0')
diff --git a/src.freebsd/bintrans/uudecode.c b/src.freebsd/bintrans/uudecode.c
index 0f388c5..a12ad51 100644
--- a/src.freebsd/bintrans/uudecode.c
+++ b/src.freebsd/bintrans/uudecode.c
@@ -93,7 +93,7 @@ main_decode(int argc, char *argv[])
 	if (strcmp(basename(argv[0]), "b64decode") == 0)
 		base64 = true;
 
-	while ((ch = getopt(argc, argv, "cimo:prs")) != -1) {
+	while ((ch = getopt(argc, argv, "+cimo:prs")) != -1) {
 		switch (ch) {
 		case 'c':
 			if (oflag || rflag)
diff --git a/src.freebsd/bintrans/uuencode.c b/src.freebsd/bintrans/uuencode.c
index a0536ad..eac8935 100644
--- a/src.freebsd/bintrans/uuencode.c
+++ b/src.freebsd/bintrans/uuencode.c
@@ -95,7 +95,7 @@ main_encode(int argc, char *argv[])
 	if (strcmp(basename(argv[0]), "b64encode") == 0)
 		base64 = 1;
 
-	while ((ch = getopt(argc, argv, "mo:rw:")) != -1) {
+	while ((ch = getopt(argc, argv, "+mo:rw:")) != -1) {
 		switch (ch) {
 		case 'm':
 			base64 = true;
diff --git a/src.freebsd/compress/compress.c b/src.freebsd/compress/compress.c
index a9f6fbb..9065315 100644
--- a/src.freebsd/compress/compress.c
+++ b/src.freebsd/compress/compress.c
@@ -80,7 +80,7 @@ main(int argc, char *argv[])
 		errx(1, "unknown program name");
 
 	bits = 0;
-	while ((ch = getopt(argc, argv, "b:cdfv")) != -1)
+	while ((ch = getopt(argc, argv, "+b:cdfv")) != -1)
 		switch(ch) {
 		case 'b':
 			bits = strtol(optarg, &p, 10);
diff --git a/src.freebsd/coreutils/basename/basename.c b/src.freebsd/coreutils/basename/basename.c
index 31d7c2a..1cfcb4c 100644
--- a/src.freebsd/coreutils/basename/basename.c
+++ b/src.freebsd/coreutils/basename/basename.c
@@ -59,7 +59,7 @@ main(int argc, char **argv)
 	suffix = NULL;
 	suffixlen = 0;
 
-	while ((ch = getopt(argc, argv, "as:")) != -1)
+	while ((ch = getopt(argc, argv, "+as:")) != -1)
 		switch(ch) {
 		case 'a':
 			aflag = 1;
diff --git a/src.freebsd/coreutils/chmod/chmod.c b/src.freebsd/coreutils/chmod/chmod.c
index e48080a..ffb4b58 100644
--- a/src.freebsd/coreutils/chmod/chmod.c
+++ b/src.freebsd/coreutils/chmod/chmod.c
@@ -69,7 +69,7 @@ main(int argc, char *argv[])
 
 	set = NULL;
 	Hflag = Lflag = Rflag = fflag = hflag = vflag = 0;
-	while ((ch = getopt(argc, argv, "LPRXfgorstuvwx")) != -1)
+	while ((ch = getopt(argc, argv, "+LPRXfgorstuvwx")) != -1)
 		switch (ch) {
 		case 'H':
 			Hflag = 1;
diff --git a/src.freebsd/coreutils/chown/chown.c b/src.freebsd/coreutils/chown/chown.c
index 35616a2..197d554 100644
--- a/src.freebsd/coreutils/chown/chown.c
+++ b/src.freebsd/coreutils/chown/chown.c
@@ -80,7 +80,7 @@ main(int argc, char **argv)
 	ischown = (strcmp(basename(argv[0]), "chown") == 0);
 
 	Hflag = Lflag = Rflag = fflag = hflag = vflag = xflag = 0;
-	while ((ch = getopt(argc, argv, "HLPRfhvx")) != -1)
+	while ((ch = getopt(argc, argv, "+HLPRfhvx")) != -1)
 		switch (ch) {
 		case 'H':
 			Hflag = 1;
diff --git a/src.freebsd/coreutils/chroot/chroot.c b/src.freebsd/coreutils/chroot/chroot.c
index 0a69af6..c9dd40f 100644
--- a/src.freebsd/coreutils/chroot/chroot.c
+++ b/src.freebsd/coreutils/chroot/chroot.c
@@ -108,7 +108,7 @@ main(int argc, char *argv[])
 	user = group = grouplist = NULL;
 	gidlist = NULL;
 	nonprivileged = false;
-	while ((ch = getopt(argc, argv, "G:g:u:n")) != -1) {
+	while ((ch = getopt(argc, argv, "+G:g:u:n")) != -1) {
 		switch(ch) {
 		case 'u':
 			user = optarg;
diff --git a/src.freebsd/coreutils/cksum/cksum.c b/src.freebsd/coreutils/cksum/cksum.c
index ce69246..ab35705 100644
--- a/src.freebsd/coreutils/cksum/cksum.c
+++ b/src.freebsd/coreutils/cksum/cksum.c
@@ -67,7 +67,7 @@ main(int argc, char **argv)
 		cfncn = crc;
 		pfncn = pcrc;
 
-		while ((ch = getopt(argc, argv, "o:")) != -1)
+		while ((ch = getopt(argc, argv, "+o:")) != -1)
 			switch (ch) {
 			case 'o':
 				if (!strcmp(optarg, "1")) {
diff --git a/src.freebsd/coreutils/comm/comm.c b/src.freebsd/coreutils/comm/comm.c
index 4dcd7cc..9a1c4cf 100644
--- a/src.freebsd/coreutils/comm/comm.c
+++ b/src.freebsd/coreutils/comm/comm.c
@@ -68,7 +68,7 @@ main(int argc, char *argv[])
 
 	flag1 = flag2 = flag3 = 1;
 
-	while ((ch = getopt(argc, argv, "123i")) != -1)
+	while ((ch = getopt(argc, argv, "+123i")) != -1)
 		switch(ch) {
 		case '1':
 			flag1 = 0;
diff --git a/src.freebsd/coreutils/csplit/csplit.c b/src.freebsd/coreutils/csplit/csplit.c
index dd64539..b04dc45 100644
--- a/src.freebsd/coreutils/csplit/csplit.c
+++ b/src.freebsd/coreutils/csplit/csplit.c
@@ -106,7 +106,7 @@ main(int argc, char *argv[])
 	kflag = sflag = 0;
 	prefix = "xx";
 	sufflen = 2;
-	while ((ch = getopt(argc, argv, "ksf:n:")) > 0) {
+	while ((ch = getopt(argc, argv, "+ksf:n:")) > 0) {
 		switch (ch) {
 		case 'f':
 			prefix = optarg;
diff --git a/src.freebsd/coreutils/cut/cut.c b/src.freebsd/coreutils/cut/cut.c
index 22f184f..6342a77 100644
--- a/src.freebsd/coreutils/cut/cut.c
+++ b/src.freebsd/coreutils/cut/cut.c
@@ -80,7 +80,7 @@ main(int argc, char *argv[])
 	dchar = '\t';			/* default delimiter is \t */
 	strcpy(dcharmb, "\t");
 
-	while ((ch = getopt(argc, argv, "b:c:d:f:snw")) != -1)
+	while ((ch = getopt(argc, argv, "+b:c:d:f:snw")) != -1)
 		switch(ch) {
 		case 'b':
 			get_list(optarg);
diff --git a/src.freebsd/coreutils/date/date.c b/src.freebsd/coreutils/date/date.c
index a5f7407..ab4de43 100644
--- a/src.freebsd/coreutils/date/date.c
+++ b/src.freebsd/coreutils/date/date.c
@@ -101,7 +101,7 @@ main(int argc, char *argv[])
 	Iflag = jflag = Rflag = 0;
 	ts.tv_sec = 0;
 	ts.tv_nsec = 0;
-	while ((ch = getopt(argc, argv, "f:I::jnRr:uv:z:")) != -1)
+	while ((ch = getopt(argc, argv, "+f:I::jnRr:uv:z:")) != -1)
 		switch((char)ch) {
 		case 'f':
 			fmt = optarg;
diff --git a/src.freebsd/coreutils/dirname/dirname.c b/src.freebsd/coreutils/dirname/dirname.c
index ff2310b..00d4159 100644
--- a/src.freebsd/coreutils/dirname/dirname.c
+++ b/src.freebsd/coreutils/dirname/dirname.c
@@ -47,7 +47,7 @@ main(int argc, char **argv)
 	if (caph_limit_stdio() < 0 || caph_enter() < 0)
 		err(1, "capsicum");
 
-	while ((ch = getopt(argc, argv, "")) != -1)
+	while ((ch = getopt(argc, argv, "+")) != -1)
 		switch(ch) {
 		case '?':
 		default:
diff --git a/src.freebsd/coreutils/expr/expr.y b/src.freebsd/coreutils/expr/expr.y
index 3dabad4..3e06f18 100644
--- a/src.freebsd/coreutils/expr/expr.y
+++ b/src.freebsd/coreutils/expr/expr.y
@@ -273,7 +273,7 @@ main(int argc, char *argv[])
 		av = argv + 1;
 		nonposix = 1;
 	} else {
-		while ((c = getopt(argc, argv, "e")) != -1) {
+		while ((c = getopt(argc, argv, "+e")) != -1) {
 			switch (c) {
 			case 'e':
 				nonposix = 1;
diff --git a/src.freebsd/coreutils/factor/factor.c b/src.freebsd/coreutils/factor/factor.c
index fa5757c..d591b83 100644
--- a/src.freebsd/coreutils/factor/factor.c
+++ b/src.freebsd/coreutils/factor/factor.c
@@ -119,7 +119,7 @@ main(int argc, char *argv[])
 	if (val == NULL)
 		errx(1, "can't initialise bignum");
 
-	while ((ch = getopt(argc, argv, "h")) != -1)
+	while ((ch = getopt(argc, argv, "+h")) != -1)
 		switch (ch) {
 		case 'h':
 			hflag++;
diff --git a/src.freebsd/coreutils/fmt/fmt.c b/src.freebsd/coreutils/fmt/fmt.c
index f4337aa..4028672 100644
--- a/src.freebsd/coreutils/fmt/fmt.c
+++ b/src.freebsd/coreutils/fmt/fmt.c
@@ -283,7 +283,7 @@ main(int argc, char *argv[])
 
 	/* 1. Grok parameters. */
 
-	while ((ch = getopt(argc, argv, "0123456789cd:hl:mnpst:w:")) != -1)
+	while ((ch = getopt(argc, argv, "+0123456789cd:hl:mnpst:w:")) != -1)
 		switch (ch) {
 		case 'c':
 			centerP = 1;
diff --git a/src.freebsd/coreutils/fold/fold.c b/src.freebsd/coreutils/fold/fold.c
index 12c4bb4..d4eba2d 100644
--- a/src.freebsd/coreutils/fold/fold.c
+++ b/src.freebsd/coreutils/fold/fold.c
@@ -62,7 +62,7 @@ main(int argc, char **argv)
 
 	width = -1;
 	previous_ch = 0;
-	while ((ch = getopt(argc, argv, "0123456789bsw:")) != -1) {
+	while ((ch = getopt(argc, argv, "+0123456789bsw:")) != -1) {
 		switch (ch) {
 		case 'b':
 			bflag = 1;
diff --git a/src.freebsd/coreutils/hostname/hostname.c b/src.freebsd/coreutils/hostname/hostname.c
index 4a03ded..f2b17af 100644
--- a/src.freebsd/coreutils/hostname/hostname.c
+++ b/src.freebsd/coreutils/hostname/hostname.c
@@ -48,7 +48,7 @@ main(int argc, char *argv[])
 
 	sflag = 0;
 	dflag = 0;
-	while ((ch = getopt(argc, argv, "fsd")) != -1)
+	while ((ch = getopt(argc, argv, "+fsd")) != -1)
 		switch (ch) {
 		case 'f':
 			/*
diff --git a/src.freebsd/coreutils/join/join.c b/src.freebsd/coreutils/join/join.c
index 2c11e85..292b84f 100644
--- a/src.freebsd/coreutils/join/join.c
+++ b/src.freebsd/coreutils/join/join.c
@@ -118,7 +118,7 @@ main(int argc, char *argv[])
 
 	aflag = vflag = 0;
 	obsolete(argv);
-	while ((ch = getopt(argc, argv, "\01a:e:j:1:2:o:t:v:")) != -1) {
+	while ((ch = getopt(argc, argv, "+\01a:e:j:1:2:o:t:v:")) != -1) {
 		switch (ch) {
 		case '\01':		/* See comment in obsolete(). */
 			aflag = 1;
diff --git a/src.freebsd/coreutils/ln/ln.cc b/src.freebsd/coreutils/ln/ln.cc
index 7557205..43d9766 100644
--- a/src.freebsd/coreutils/ln/ln.cc
+++ b/src.freebsd/coreutils/ln/ln.cc
@@ -76,7 +76,7 @@ main(int argc, char *argv[])
 	 * simplified.
 	 */
 	if (strcmp(getprogname(), "link") == 0) {
-		while (getopt(argc, argv, "") != -1)
+		while (getopt(argc, argv, "+") != -1)
 			link_usage();
 		argc -= optind;
 		argv += optind;
@@ -92,7 +92,7 @@ main(int argc, char *argv[])
 		exit(linkit(argv[0], argv[1], false));
 	}
 
-	while ((ch = getopt(argc, argv, "FLPTfhinst:vwr")) != -1)
+	while ((ch = getopt(argc, argv, "+FLPTfhinst:vwr")) != -1)
 		switch (ch) {
 		case 'F':
 			Fflag = true;
diff --git a/src.freebsd/coreutils/mkdir/mkdir.c b/src.freebsd/coreutils/mkdir/mkdir.c
index e67c99c..f5dac74 100644
--- a/src.freebsd/coreutils/mkdir/mkdir.c
+++ b/src.freebsd/coreutils/mkdir/mkdir.c
@@ -56,7 +56,7 @@ main(int argc, char *argv[])
 
 	omode = pflag = 0;
 	mode = NULL;
-	while ((ch = getopt(argc, argv, "m:pv")) != -1)
+	while ((ch = getopt(argc, argv, "+m:pv")) != -1)
 		switch(ch) {
 		case 'm':
 			mode = optarg;
diff --git a/src.freebsd/coreutils/mkfifo/mkfifo.c b/src.freebsd/coreutils/mkfifo/mkfifo.c
index fb461ef..2da3695 100644
--- a/src.freebsd/coreutils/mkfifo/mkfifo.c
+++ b/src.freebsd/coreutils/mkfifo/mkfifo.c
@@ -55,7 +55,7 @@ main(int argc, char *argv[])
 	mode_t fifomode;
 	int ch, exitval;
 
-	while ((ch = getopt(argc, argv, "m:")) != -1)
+	while ((ch = getopt(argc, argv, "+m:")) != -1)
 		switch(ch) {
 		case 'm':
 			f_mode = 1;
diff --git a/src.freebsd/coreutils/mv/mv.c b/src.freebsd/coreutils/mv/mv.c
index 4b5d531..234b43b 100644
--- a/src.freebsd/coreutils/mv/mv.c
+++ b/src.freebsd/coreutils/mv/mv.c
@@ -80,7 +80,7 @@ main(int argc, char *argv[])
 	int ch, rval;
 	const char *target;
 
-	while ((ch = getopt(argc, argv, "Tfhint:vx")) != -1)
+	while ((ch = getopt(argc, argv, "+Tfhint:vx")) != -1)
 		switch (ch) {
 		case 'h':
 			hflg = 1;
diff --git a/src.freebsd/coreutils/nice/nice.c b/src.freebsd/coreutils/nice/nice.c
index d6fda3f..b9c5639 100644
--- a/src.freebsd/coreutils/nice/nice.c
+++ b/src.freebsd/coreutils/nice/nice.c
@@ -59,7 +59,7 @@ main(int argc, char *argv[])
 		if (asprintf(&argv[1], "-n%s", argv[1] + 1) < 0)
 			err(1, "asprintf");
 
-	while ((ch = getopt(argc, argv, "n:")) != -1) {
+	while ((ch = getopt(argc, argv, "+n:")) != -1) {
 		switch (ch) {
 		case 'n':
 			errno = 0;
diff --git a/src.freebsd/coreutils/nl/nl.c b/src.freebsd/coreutils/nl/nl.c
index 05837e6..14c2a51 100644
--- a/src.freebsd/coreutils/nl/nl.c
+++ b/src.freebsd/coreutils/nl/nl.c
@@ -135,7 +135,7 @@ main(int argc, char *argv[])
 
 	(void)setlocale(LC_ALL, "");
 
-	while ((c = getopt(argc, argv, "pb:d:f:h:i:l:n:s:v:w:")) != -1) {
+	while ((c = getopt(argc, argv, "+pb:d:f:h:i:l:n:s:v:w:")) != -1) {
 		switch (c) {
 		case 'p':
 			restart = 0;
diff --git a/src.freebsd/coreutils/nohup/nohup.c b/src.freebsd/coreutils/nohup/nohup.c
index 74ca836..d9e449f 100644
--- a/src.freebsd/coreutils/nohup/nohup.c
+++ b/src.freebsd/coreutils/nohup/nohup.c
@@ -60,7 +60,7 @@ main(int argc, char *argv[])
 {
 	int exit_status;
 
-	while (getopt(argc, argv, "") != -1)
+	while (getopt(argc, argv, "+") != -1)
 		usage();
 	argc -= optind;
 	argv += optind;
diff --git a/src.freebsd/coreutils/paste/paste.c b/src.freebsd/coreutils/paste/paste.c
index 8d7901f..5de6630 100644
--- a/src.freebsd/coreutils/paste/paste.c
+++ b/src.freebsd/coreutils/paste/paste.c
@@ -66,7 +66,7 @@ main(int argc, char *argv[])
 	setlocale(LC_CTYPE, "");
 
 	seq = 0;
-	while ((ch = getopt(argc, argv, "d:s")) != -1)
+	while ((ch = getopt(argc, argv, "+d:s")) != -1)
 		switch(ch) {
 		case 'd':
 			arg = optarg;
diff --git a/src.freebsd/coreutils/pathchk/pathchk.c b/src.freebsd/coreutils/pathchk/pathchk.c
index 4b9dddf..ae670b7 100644
--- a/src.freebsd/coreutils/pathchk/pathchk.c
+++ b/src.freebsd/coreutils/pathchk/pathchk.c
@@ -58,7 +58,7 @@ main(int argc, char *argv[])
 	int ch, rval;
 	const char *arg;
 
-	while ((ch = getopt(argc, argv, "pP")) > 0) {
+	while ((ch = getopt(argc, argv, "+pP")) > 0) {
 		switch (ch) {
 		case 'p':
 			pflag = 1;
diff --git a/src.freebsd/coreutils/pr/pr.c b/src.freebsd/coreutils/pr/pr.c
index 16461d4..833aa73 100644
--- a/src.freebsd/coreutils/pr/pr.c
+++ b/src.freebsd/coreutils/pr/pr.c
@@ -1609,7 +1609,7 @@ setup(int argc, char *argv[])
 		}
 	} else
 		err = stderr;
-	while ((c = egetopt(argc, argv, "#adFfmrte?h:i?L:l:n?o:ps?w:")) != -1) {
+	while ((c = egetopt(argc, argv, "+#adFfmrte?h:i?L:l:n?o:ps?w:")) != -1) {
 		switch (c) {
 		case '+':
 			if ((pgnm = atoi(eoptarg)) < 1) {
diff --git a/src.freebsd/coreutils/printenv/printenv.c b/src.freebsd/coreutils/printenv/printenv.c
index 43dbdb0..30f2418 100644
--- a/src.freebsd/coreutils/printenv/printenv.c
+++ b/src.freebsd/coreutils/printenv/printenv.c
@@ -57,7 +57,7 @@ main(int argc, char *argv[])
 	if (caph_limit_stdio() < 0 || caph_enter() < 0)
 		err(1, "capsicum");
 
-	while ((ch = getopt(argc, argv, "")) != -1)
+	while ((ch = getopt(argc, argv, "+")) != -1)
 		switch(ch) {
 		case '?':
 		default:
diff --git a/src.freebsd/coreutils/printf/printf.c b/src.freebsd/coreutils/printf/printf.c
index 18e7a4d..a25d3f3 100644
--- a/src.freebsd/coreutils/printf/printf.c
+++ b/src.freebsd/coreutils/printf/printf.c
@@ -108,7 +108,7 @@ main(int argc, char *argv[])
 	argc -= argptr - argv;
 	argv = argptr;
 #else
-	while ((ch = getopt(argc, argv, "")) != -1)
+	while ((ch = getopt(argc, argv, "+")) != -1)
 		switch (ch) {
 		case '?':
 		default:
diff --git a/src.freebsd/coreutils/pwd/pwd.c b/src.freebsd/coreutils/pwd/pwd.c
index 8e055a8..71935b1 100644
--- a/src.freebsd/coreutils/pwd/pwd.c
+++ b/src.freebsd/coreutils/pwd/pwd.c
@@ -51,7 +51,7 @@ main(int argc, char *argv[])
 	char *p;
 
 	physical = 1;
-	while ((ch = getopt(argc, argv, "LP")) != -1)
+	while ((ch = getopt(argc, argv, "+LP")) != -1)
 		switch (ch) {
 		case 'L':
 			physical = 0;
diff --git a/src.freebsd/coreutils/realpath/realpath.c b/src.freebsd/coreutils/realpath/realpath.c
index 61263e5..06dcdff 100644
--- a/src.freebsd/coreutils/realpath/realpath.c
+++ b/src.freebsd/coreutils/realpath/realpath.c
@@ -48,7 +48,7 @@ main(int argc, char *argv[])
 	int ch, qflag, rval;
 
 	qflag = 0;
-	while ((ch = getopt(argc, argv, "q")) != -1) {
+	while ((ch = getopt(argc, argv, "+q")) != -1) {
 		switch (ch) {
 		case 'q':
 			qflag = 1;
diff --git a/src.freebsd/coreutils/rm/rm.c b/src.freebsd/coreutils/rm/rm.c
index 708c817..6cbadb8 100644
--- a/src.freebsd/coreutils/rm/rm.c
+++ b/src.freebsd/coreutils/rm/rm.c
@@ -97,7 +97,7 @@ main(int argc, char *argv[])
 	}
 
 	rflag = xflag = 0;
-	while ((ch = getopt(argc, argv, "dfiIPRrvx")) != -1)
+	while ((ch = getopt(argc, argv, "+dfiIPRrvx")) != -1)
 		switch(ch) {
 		case 'd':
 			dflag = 1;
diff --git a/src.freebsd/coreutils/rmdir/rmdir.c b/src.freebsd/coreutils/rmdir/rmdir.c
index 115ae01..646a98d 100644
--- a/src.freebsd/coreutils/rmdir/rmdir.c
+++ b/src.freebsd/coreutils/rmdir/rmdir.c
@@ -47,7 +47,7 @@ main(int argc, char *argv[])
 {
 	int ch, errors;
 
-	while ((ch = getopt(argc, argv, "pv")) != -1)
+	while ((ch = getopt(argc, argv, "+pv")) != -1)
 		switch(ch) {
 		case 'p':
 			pflag = 1;
diff --git a/src.freebsd/coreutils/sleep/sleep.c b/src.freebsd/coreutils/sleep/sleep.c
index c6c9c81..4295719 100644
--- a/src.freebsd/coreutils/sleep/sleep.c
+++ b/src.freebsd/coreutils/sleep/sleep.c
@@ -97,7 +97,7 @@ main(int argc, char *argv[])
 	if (caph_limit_stdio() < 0 || caph_enter() < 0)
 		err(1, "capsicum");
 
-	while (getopt(argc, argv, "") != -1)
+	while (getopt(argc, argv, "+") != -1)
 		usage();
 	argc -= optind;
 	argv += optind;
diff --git a/src.freebsd/coreutils/split/split.c b/src.freebsd/coreutils/split/split.c
index 63cf055..6743158 100644
--- a/src.freebsd/coreutils/split/split.c
+++ b/src.freebsd/coreutils/split/split.c
@@ -85,7 +85,7 @@ main(int argc, char **argv)
 	setlocale(LC_ALL, "");
 
 	dflag = false;
-	while ((ch = getopt(argc, argv, "0::1::2::3::4::5::6::7::8::9::a:b:cdl:n:p:")) != -1)
+	while ((ch = getopt(argc, argv, "+0::1::2::3::4::5::6::7::8::9::a:b:cdl:n:p:")) != -1)
 		switch (ch) {
 		case '0': case '1': case '2': case '3': case '4':
 		case '5': case '6': case '7': case '8': case '9':
diff --git a/src.freebsd/coreutils/stat/stat.c b/src.freebsd/coreutils/stat/stat.c
index f3601cd..5e750d6 100644
--- a/src.freebsd/coreutils/stat/stat.c
+++ b/src.freebsd/coreutils/stat/stat.c
@@ -232,13 +232,13 @@ main(int argc, char *argv[])
 
 	if (strcmp(getprogname(), "readlink") == 0) {
 		am_readlink = 1;
-		options = "fn";
+		options = "+fn";
 		synopsis = "[-fn] [file ...]";
 		statfmt = "%Y";
 		fmtchar = 'f';
 		quiet = 1;
 	} else {
-		options = "Ff:hLlnqrst:x";
+		options = "+Ff:hLlnqrst:x";
 		synopsis = "[-FHhLnq] [-f format | -l | -r | -s | -x] "
 		    "[-t timefmt] [file|handle ...]";
 	}
diff --git a/src.freebsd/coreutils/stdbuf/stdbuf.c b/src.freebsd/coreutils/stdbuf/stdbuf.c
index 1268c74..b70fcdd 100644
--- a/src.freebsd/coreutils/stdbuf/stdbuf.c
+++ b/src.freebsd/coreutils/stdbuf/stdbuf.c
@@ -70,7 +70,7 @@ main(int argc, char *argv[])
 	int i;
 
 	ibuf = obuf = ebuf = NULL;
-	while ((i = getopt(argc, argv, "e:i:o:")) != -1) {
+	while ((i = getopt(argc, argv, "+e:i:o:")) != -1) {
 		switch (i) {
 		case 'e':
 			ebuf = optarg;
diff --git a/src.freebsd/coreutils/stty/stty.c b/src.freebsd/coreutils/stty/stty.c
index e1a16ea..5aa505e 100644
--- a/src.freebsd/coreutils/stty/stty.c
+++ b/src.freebsd/coreutils/stty/stty.c
@@ -58,7 +58,7 @@ main(int argc, char *argv[])
 	opterr = 0;
 	while (optind < argc &&
 	    strspn(argv[optind], "-aefg") == strlen(argv[optind]) &&
-	    (ch = getopt(argc, argv, "aef:g")) != -1)
+	    (ch = getopt(argc, argv, "+aef:g")) != -1)
 		switch(ch) {
 		case 'a':		/* undocumented: POSIX compatibility */
 			fmt = POSIX;
diff --git a/src.freebsd/coreutils/tee/tee.c b/src.freebsd/coreutils/tee/tee.c
index be8c4fe..563925c 100644
--- a/src.freebsd/coreutils/tee/tee.c
+++ b/src.freebsd/coreutils/tee/tee.c
@@ -67,7 +67,7 @@ main(int argc, char *argv[])
 #define	BSIZE (8 * 1024)
 
 	append = 0;
-	while ((ch = getopt(argc, argv, "ai")) != -1)
+	while ((ch = getopt(argc, argv, "+ai")) != -1)
 		switch((char)ch) {
 		case 'a':
 			append = 1;
diff --git a/src.freebsd/coreutils/touch/touch.c b/src.freebsd/coreutils/touch/touch.c
index a883522..72bc745 100644
--- a/src.freebsd/coreutils/touch/touch.c
+++ b/src.freebsd/coreutils/touch/touch.c
@@ -67,7 +67,7 @@ main(int argc, char *argv[])
 	ts[0].tv_sec = ts[1].tv_sec = 0;
 	ts[0].tv_nsec = ts[1].tv_nsec = UTIME_NOW;
 
-	while ((ch = getopt(argc, argv, "A:acd:fhmr:t:")) != -1)
+	while ((ch = getopt(argc, argv, "+A:acd:fhmr:t:")) != -1)
 		switch(ch) {
 		case 'A':
 			Aflag = timeoffset(optarg);
diff --git a/src.freebsd/coreutils/tr/tr.c b/src.freebsd/coreutils/tr/tr.c
index ba43e2a..b5f81e3 100644
--- a/src.freebsd/coreutils/tr/tr.c
+++ b/src.freebsd/coreutils/tr/tr.c
@@ -76,7 +76,7 @@ main(int argc, char **argv)
 		err(1, "unable to enter capability mode");
 
 	Cflag = cflag = dflag = sflag = 0;
-	while ((optc = getopt(argc, argv, "Ccdsu")) != -1)
+	while ((optc = getopt(argc, argv, "+Ccdsu")) != -1)
 		switch(optc) {
 		case 'C':
 			Cflag = 1;
diff --git a/src.freebsd/coreutils/truncate/truncate.c b/src.freebsd/coreutils/truncate/truncate.c
index 6694231..c602867 100644
--- a/src.freebsd/coreutils/truncate/truncate.c
+++ b/src.freebsd/coreutils/truncate/truncate.c
@@ -72,7 +72,7 @@ main(int argc, char **argv)
 	do_truncate = 1;
 	error = r = 0;
 	rname = NULL;
-	while ((ch = getopt(argc, argv, "cdr:s:o:l:")) != -1)
+	while ((ch = getopt(argc, argv, "+cdr:s:o:l:")) != -1)
 		switch (ch) {
 		case 'c':
 			no_create = 1;
diff --git a/src.freebsd/coreutils/tsort/tsort.c b/src.freebsd/coreutils/tsort/tsort.c
index b55c494..f18734d 100644
--- a/src.freebsd/coreutils/tsort/tsort.c
+++ b/src.freebsd/coreutils/tsort/tsort.c
@@ -183,7 +183,7 @@ main(int argc, char *argv[])
 	BUF bufs[2];
 
 	fp = NULL;
-	while ((ch = getopt(argc, argv, "dlq")) != -1)
+	while ((ch = getopt(argc, argv, "+dlq")) != -1)
 		switch (ch) {
 		case 'd':
 			debug = 1;
diff --git a/src.freebsd/coreutils/tty/tty.c b/src.freebsd/coreutils/tty/tty.c
index fe6ac24..9756181 100644
--- a/src.freebsd/coreutils/tty/tty.c
+++ b/src.freebsd/coreutils/tty/tty.c
@@ -43,7 +43,7 @@ main(int argc, char *argv[])
 	char *t;
 
 	sflag = 0;
-	while ((ch = getopt(argc, argv, "s")) != -1)
+	while ((ch = getopt(argc, argv, "+s")) != -1)
 		switch (ch) {
 		case 's':
 			sflag = 1;
diff --git a/src.freebsd/coreutils/uname/uname.c b/src.freebsd/coreutils/uname/uname.c
index 8db2197..ea01f01 100644
--- a/src.freebsd/coreutils/uname/uname.c
+++ b/src.freebsd/coreutils/uname/uname.c
@@ -86,7 +86,7 @@ main(int argc, char *argv[])
 	setup_get();
 	flags = 0;
 
-	while ((ch = getopt(argc, argv, "abiKmnoprsUv")) != -1)
+	while ((ch = getopt(argc, argv, "+abiKmnoprsUv")) != -1)
 		switch(ch) {
 		case 'a':
 			flags |= (MFLAG | NFLAG | RFLAG | SFLAG | VFLAG);
diff --git a/src.freebsd/coreutils/unexpand/unexpand.c b/src.freebsd/coreutils/unexpand/unexpand.c
index e89dd0d..199e649 100644
--- a/src.freebsd/coreutils/unexpand/unexpand.c
+++ b/src.freebsd/coreutils/unexpand/unexpand.c
@@ -62,7 +62,7 @@ main(int argc, char *argv[])
 
 	nstops = 1;
 	tabstops[0] = 8;
-	while ((ch = getopt(argc, argv, "at:")) != -1) {
+	while ((ch = getopt(argc, argv, "+at:")) != -1) {
 		switch (ch) {
 		case 'a':	/* Un-expand all spaces, not just leading. */
 			all = 1;
diff --git a/src.freebsd/coreutils/wc/wc.c b/src.freebsd/coreutils/wc/wc.c
index bcb6675..1614e78 100644
--- a/src.freebsd/coreutils/wc/wc.c
+++ b/src.freebsd/coreutils/wc/wc.c
@@ -96,7 +96,7 @@ main(int argc, char *argv[])
 	if (argc < 0)
 		exit(EXIT_FAILURE);
 
-	while ((ch = getopt(argc, argv, "clmwL")) != -1)
+	while ((ch = getopt(argc, argv, "+clmwL")) != -1)
 		switch((char)ch) {
 		case 'l':
 			doline = true;
diff --git a/src.freebsd/coreutils/who/who.c b/src.freebsd/coreutils/who/who.c
index f71adee..7bc52cd 100644
--- a/src.freebsd/coreutils/who/who.c
+++ b/src.freebsd/coreutils/who/who.c
@@ -69,7 +69,7 @@ main(int argc, char *argv[])
 
 	setlocale(LC_TIME, "");
 
-	while ((ch = getopt(argc, argv, "HTabmqsu")) != -1) {
+	while ((ch = getopt(argc, argv, "+HTabmqsu")) != -1) {
 		switch (ch) {
 		case 'H':		/* Write column headings */
 			Hflag = 1;
diff --git a/src.freebsd/coreutils/xinstall/xinstall.c b/src.freebsd/coreutils/xinstall/xinstall.c
index 1b65231..4d62c27 100644
--- a/src.freebsd/coreutils/xinstall/xinstall.c
+++ b/src.freebsd/coreutils/xinstall/xinstall.c
@@ -162,7 +162,7 @@ static int
 do_getopt(int argc, char **argv)
 {
 	if (!gnumode)
-		return getopt(argc, argv, "B:bCcD:dg:h:l:M:m:o:pSsT:Uv");
+		return getopt(argc, argv, "+B:bCcD:dg:h:l:M:m:o:pSsT:Uv");
 	/* use getopt_long to get argument interposition */
 	return getopt_long(argc, argv, "B:bCcDdg:l:m:o:pSst:Tv", NULL, NULL);
 }
diff --git a/src.freebsd/ed/main.c b/src.freebsd/ed/main.c
index 241d9f1..cd19fdb 100644
--- a/src.freebsd/ed/main.c
+++ b/src.freebsd/ed/main.c
@@ -96,7 +96,7 @@ main(volatile int argc, char ** volatile argv)
 
 	red = (n = strlen(argv[0])) > 2 && argv[0][n - 3] == 'r';
 top:
-	while ((c = getopt(argc, argv, "p:sx")) != -1)
+	while ((c = getopt(argc, argv, "+p:sx")) != -1)
 		switch(c) {
 		case 'p':				/* set prompt */
 			prompt = optarg;
diff --git a/src.freebsd/findutils/locate/code/locate.code.c b/src.freebsd/findutils/locate/code/locate.code.c
index 624ed44..f8f809d 100644
--- a/src.freebsd/findutils/locate/code/locate.code.c
+++ b/src.freebsd/findutils/locate/code/locate.code.c
@@ -107,7 +107,7 @@ main(int argc, char *argv[])
 	u_int i, j;
 	FILE *fp;
 
-	while ((ch = getopt(argc, argv, "")) != -1)
+	while ((ch = getopt(argc, argv, "+")) != -1)
 		switch(ch) {
 		default:
 			usage();
diff --git a/src.freebsd/findutils/locate/locate/locate.c b/src.freebsd/findutils/locate/locate/locate.c
index 60cbd73..8670483 100644
--- a/src.freebsd/findutils/locate/locate/locate.c
+++ b/src.freebsd/findutils/locate/locate/locate.c
@@ -124,7 +124,7 @@ main(int argc, char **argv)
 #endif
 	(void) setlocale(LC_ALL, "");
 
-        while ((ch = getopt(argc, argv, "0Scd:il:ms")) != -1)
+        while ((ch = getopt(argc, argv, "+0Scd:il:ms")) != -1)
                 switch(ch) {
                 case '0':	/* 'find -print0' style */
 			separator = '\0';
diff --git a/src.freebsd/gencat/gencat.c b/src.freebsd/gencat/gencat.c
index dc043cd..710b79a 100644
--- a/src.freebsd/gencat/gencat.c
+++ b/src.freebsd/gencat/gencat.c
@@ -161,9 +161,9 @@ main(int argc, char **argv)
 #define DEPRECATEDMSG	1
 
 #ifdef DEPRECATEDMSG
-	while ((c = getopt(argc, argv, "new")) != -1) {
+	while ((c = getopt(argc, argv, "+new")) != -1) {
 #else
-	while ((c = getopt(argc, argv, "")) != -1) {
+	while ((c = getopt(argc, argv, "+")) != -1) {
 #endif
 		switch (c) {
 #ifdef DEPRECATEDMSG
diff --git a/src.freebsd/jot/jot.c b/src.freebsd/jot/jot.c
index b316a86..a667dc5 100644
--- a/src.freebsd/jot/jot.c
+++ b/src.freebsd/jot/jot.c
@@ -113,7 +113,7 @@ main(int argc, char **argv)
 	if (caph_enter() < 0)
 		err(1, "unable to enter capability mode");
 
-	while ((ch = getopt(argc, argv, "b:cnp:rs:w:")) != -1)
+	while ((ch = getopt(argc, argv, "+b:cnp:rs:w:")) != -1)
 		switch (ch) {
 		case 'b':
 			boring = true;
diff --git a/src.freebsd/miscutils/col/col.c b/src.freebsd/miscutils/col/col.c
index e5bf062..c854e9f 100644
--- a/src.freebsd/miscutils/col/col.c
+++ b/src.freebsd/miscutils/col/col.c
@@ -135,7 +135,7 @@ main(int argc, char **argv)
 
 	max_bufd_lines = 256;
 	compress_spaces = 1;		/* compress spaces into tabs */
-	while ((opt = getopt(argc, argv, "bfhl:px")) != -1)
+	while ((opt = getopt(argc, argv, "+bfhl:px")) != -1)
 		switch (opt) {
 		case 'b':		/* do not output backspaces */
 			no_backspaces = 1;
diff --git a/src.freebsd/miscutils/colrm/colrm.c b/src.freebsd/miscutils/colrm/colrm.c
index 0c6b6ad..07c7859 100644
--- a/src.freebsd/miscutils/colrm/colrm.c
+++ b/src.freebsd/miscutils/colrm/colrm.c
@@ -61,7 +61,7 @@ main(int argc, char *argv[])
 	if (caph_limit_stdio() < 0 || caph_enter() < 0)
 		err(EXIT_FAILURE, "capsicum");
 
-	while ((ch = getopt(argc, argv, "")) != -1)
+	while ((ch = getopt(argc, argv, "+")) != -1)
 		switch(ch) {
 		case '?':
 		default:
diff --git a/src.freebsd/miscutils/column/column.c b/src.freebsd/miscutils/column/column.c
index aa9e38d..3ad1a3e 100644
--- a/src.freebsd/miscutils/column/column.c
+++ b/src.freebsd/miscutils/column/column.c
@@ -82,7 +82,7 @@ main(int argc, char **argv)
 		termwidth = win.ws_col;
 
 	tflag = xflag = 0;
-	while ((ch = getopt(argc, argv, "c:l:s:tx")) != -1)
+	while ((ch = getopt(argc, argv, "+c:l:s:tx")) != -1)
 		switch(ch) {
 		case 'c':
 			termwidth = strtonum(optarg, 0, INT_MAX, &errstr);
diff --git a/src.freebsd/miscutils/hexdump/hexsyntax.c b/src.freebsd/miscutils/hexdump/hexsyntax.c
index de6b384..fdb6545 100644
--- a/src.freebsd/miscutils/hexdump/hexsyntax.c
+++ b/src.freebsd/miscutils/hexdump/hexsyntax.c
@@ -55,7 +55,7 @@ newsyntax(int argc, char ***argvp)
 		add("\"%08.8_ax  \" 8/1 \"%02x \" \"  \" 8/1 \"%02x \" ");
 		add("\"  |\" 16/1 \"%_p\" \"|\\n\"");
 	}
-	while ((ch = getopt(argc, argv, "bcCde:f:n:os:vx")) != -1)
+	while ((ch = getopt(argc, argv, "+bcCde:f:n:os:vx")) != -1)
 		switch (ch) {
 		case 'b':
 			add("\"%07.7_Ax\n\"");
diff --git a/src.freebsd/miscutils/hexdump/odsyntax.c b/src.freebsd/miscutils/hexdump/odsyntax.c
index c292fba..351905d 100644
--- a/src.freebsd/miscutils/hexdump/odsyntax.c
+++ b/src.freebsd/miscutils/hexdump/odsyntax.c
@@ -68,7 +68,7 @@ oldsyntax(int argc, char ***argvp)
 
 	odmode = 1;
 	argv = *argvp;
-	while ((ch = getopt(argc, argv, "A:aBbcDdeFfHhIij:LlN:Oost:vXx")) != -1)
+	while ((ch = getopt(argc, argv, "+A:aBbcDdeFfHhIij:LlN:Oost:vXx")) != -1)
 		switch (ch) {
 		case 'A':
 			switch (*optarg) {
diff --git a/src.freebsd/miscutils/logger/logger.c b/src.freebsd/miscutils/logger/logger.c
index 53d9144..defff17 100644
--- a/src.freebsd/miscutils/logger/logger.c
+++ b/src.freebsd/miscutils/logger/logger.c
@@ -102,7 +102,7 @@ main(int argc, char *argv[])
 	pristr = NULL;
 	logflags = 0;
 	unsetenv("TZ");
-	while ((ch = getopt(argc, argv, "46Af:H:h:iP:p:S:st:")) != -1)
+	while ((ch = getopt(argc, argv, "+46Af:H:h:iP:p:S:st:")) != -1)
 		switch((char)ch) {
 		case '4':
 			family = PF_INET;
diff --git a/src.freebsd/miscutils/mesg/mesg.c b/src.freebsd/miscutils/mesg/mesg.c
index 34c781a..a4596d9 100644
--- a/src.freebsd/miscutils/mesg/mesg.c
+++ b/src.freebsd/miscutils/mesg/mesg.c
@@ -53,7 +53,7 @@ main(int argc, char *argv[])
 	char *tty;
 	int ch;
 
-	while ((ch = getopt(argc, argv, "")) != -1)
+	while ((ch = getopt(argc, argv, "+")) != -1)
 		switch (ch) {
 		case '?':
 		default:
diff --git a/src.freebsd/miscutils/ncal/ncal.c b/src.freebsd/miscutils/ncal/ncal.c
index e6279e1..024bb98 100644
--- a/src.freebsd/miscutils/ncal/ncal.c
+++ b/src.freebsd/miscutils/ncal/ncal.c
@@ -259,7 +259,7 @@ main(int argc, char *argv[])
 
 	before = after = -1;
 
-	while ((ch = getopt(argc, argv, "3A:B:Cd:eH:hjJm:Nops:wyM")) != -1)
+	while ((ch = getopt(argc, argv, "+3A:B:Cd:eH:hjJm:Nops:wyM")) != -1)
 		switch (ch) {
 		case '3':
 			flag_3months = 1;
diff --git a/src.freebsd/miscutils/rev/rev.c b/src.freebsd/miscutils/rev/rev.c
index 8711d71..561347e 100644
--- a/src.freebsd/miscutils/rev/rev.c
+++ b/src.freebsd/miscutils/rev/rev.c
@@ -98,7 +98,7 @@ main(int argc, char *argv[])
 
 	setlocale(LC_ALL, "");
 
-	while ((ch = getopt(argc, argv, "")) != -1)
+	while ((ch = getopt(argc, argv, "+")) != -1)
 		switch(ch) {
 		case '?':
 		default:
diff --git a/src.freebsd/miscutils/script/script.c b/src.freebsd/miscutils/script/script.c
index 12e5398..b8bbd54 100644
--- a/src.freebsd/miscutils/script/script.c
+++ b/src.freebsd/miscutils/script/script.c
@@ -132,7 +132,7 @@ main(int argc, char *argv[])
 	sigemptyset(&selmask);
 	pselmask = NULL;
 
-	while ((ch = getopt(argc, argv, "adeFkpqrT:t:w")) != -1)
+	while ((ch = getopt(argc, argv, "+adeFkpqrT:t:w")) != -1)
 		switch (ch) {
 		case 'a':
 			aflg = 1;
diff --git a/src.freebsd/miscutils/ul/ul.c b/src.freebsd/miscutils/ul/ul.c
index 8c35536..c74eeb8 100644
--- a/src.freebsd/miscutils/ul/ul.c
+++ b/src.freebsd/miscutils/ul/ul.c
@@ -104,7 +104,7 @@ main(int argc, char **argv)
 	termtype = getenv("TERM");
 	if (termtype == NULL || (argv[0][0] == 'c' && !isatty(1)))
 		termtype = "lpr";
-	while ((c = getopt(argc, argv, "it:T:")) != -1)
+	while ((c = getopt(argc, argv, "+it:T:")) != -1)
 		switch (c) {
 		case 't':
 		case 'T': /* for nroff compatibility */
diff --git a/src.freebsd/miscutils/wall/wall.c b/src.freebsd/miscutils/wall/wall.c
index 60e555b..8782fb5 100644
--- a/src.freebsd/miscutils/wall/wall.c
+++ b/src.freebsd/miscutils/wall/wall.c
@@ -97,7 +97,7 @@ main(int argc, char *argv[])
 
 	(void)setlocale(LC_CTYPE, "");
 
-	while ((ch = getopt(argc, argv, "g:n")) != -1)
+	while ((ch = getopt(argc, argv, "+g:n")) != -1)
 		switch (ch) {
 		case 'n':
 			/* undoc option for shutdown: suppress banner */
diff --git a/src.freebsd/miscutils/whereis/whereis.c b/src.freebsd/miscutils/whereis/whereis.c
index 527047b..4c41219 100644
--- a/src.freebsd/miscutils/whereis/whereis.c
+++ b/src.freebsd/miscutils/whereis/whereis.c
@@ -105,7 +105,7 @@ scanopts(int argc, char **argv)
 	int c, i;
 	ccharp **dirlist;
 
-	while ((c = getopt(argc, argv, "BMSabfmqsux")) != -1)
+	while ((c = getopt(argc, argv, "+BMSabfmqsux")) != -1)
 		switch (c) {
 		case 'B':
 			dirlist = &bindirs;
diff --git a/src.freebsd/miscutils/write/write.c b/src.freebsd/miscutils/write/write.c
index ead46bf..f29b956 100644
--- a/src.freebsd/miscutils/write/write.c
+++ b/src.freebsd/miscutils/write/write.c
@@ -127,7 +127,7 @@ main(int argc, char **argv)
 	if (caph_enter() < 0)
 		err(1, "cap_enter");
 
-	while (getopt(argc, argv, "") != -1)
+	while (getopt(argc, argv, "+") != -1)
 		usage();
 	argc -= optind;
 	argv += optind;
diff --git a/src.freebsd/nvi/common/main.c b/src.freebsd/nvi/common/main.c
index 26fb482..8f6bd2c 100644
--- a/src.freebsd/nvi/common/main.c
+++ b/src.freebsd/nvi/common/main.c
@@ -102,7 +102,7 @@ editor(GS *gp, int argc, char *argv[])
 #ifdef DEBUG
 	while ((ch = getopt(argc, argv, "c:D:eFlRrSsT:t:vw:")) != EOF)
 #else
-	while ((ch = getopt(argc, argv, "c:eFlRrSst:vw:")) != EOF)
+	while ((ch = getopt(argc, argv, "+c:eFlRrSst:vw:")) != EOF)
 #endif
 		switch (ch) {
 		case 'c':		/* Run the command. */
diff --git a/src.freebsd/su/su.c b/src.freebsd/su/su.c
index de1f447..9cf31db 100644
--- a/src.freebsd/su/su.c
+++ b/src.freebsd/su/su.c
@@ -168,7 +168,7 @@ main(int argc, char *argv[])
 	asme = asthem = statusp = 0;
 	user = "root";
 
-	while ((ch = getopt(argc, argv, "lm")) != -1)
+	while ((ch = getopt(argc, argv, "+lm")) != -1)
 		switch ((char)ch) {
 		case 'l':
 			asme = 0;
diff --git a/src.freebsd/telnet/telnet/main.c b/src.freebsd/telnet/telnet/main.c
index 622866e..5fb106b 100644
--- a/src.freebsd/telnet/telnet/main.c
+++ b/src.freebsd/telnet/telnet/main.c
@@ -152,7 +152,7 @@ main(int argc, char *argv[])
 #define IPSECOPT
 #endif
 	while ((ch = getopt(argc, argv,
-			    "468B:EKLNQS:X:acde:fFk:l:n:rs:uxy" IPSECOPT)) != -1)
+			    "+468B:EKLNQS:X:acde:fFk:l:n:rs:uxy" IPSECOPT)) != -1)
 #undef IPSECOPT
 	{
 		switch(ch) {
diff --git a/src.freebsd/tip/tip/cu.c b/src.freebsd/tip/tip/cu.c
index 6811432..1f7d7ac 100644
--- a/src.freebsd/tip/tip/cu.c
+++ b/src.freebsd/tip/tip/cu.c
@@ -71,7 +71,7 @@ cumain(int argc, char *argv[])
 		}
 	}
 
-	while ((ch = getopt(argc, argv, "a:l:s:htoe")) != -1) {
+	while ((ch = getopt(argc, argv, "+a:l:s:htoe")) != -1) {
 		switch (ch) {
 		case 'a':
 			CU = optarg;
diff --git a/src.freebsd/unvis/unvis.c b/src.freebsd/unvis/unvis.c
index 52614f8..c62ee86 100644
--- a/src.freebsd/unvis/unvis.c
+++ b/src.freebsd/unvis/unvis.c
@@ -56,7 +56,7 @@ main(int argc, char *argv[])
 	FILE *fp;
 	int ch, eflags = 0;
 
-	while ((ch = getopt(argc, argv, "eHhm")) != -1)
+	while ((ch = getopt(argc, argv, "+eHhm")) != -1)
 		switch((char)ch) {
 		case 'e':
 			eflags |= VIS_NOESCAPE;
diff --git a/src.freebsd/vis/vis.c b/src.freebsd/vis/vis.c
index 6e3ea11..f46f3d9 100644
--- a/src.freebsd/vis/vis.c
+++ b/src.freebsd/vis/vis.c
@@ -70,7 +70,7 @@ main(int argc, char *argv[])
 	int ch;
 	int rval;
 
-	while ((ch = getopt(argc, argv, "bcde:F:fhlMmNnoSstw")) != -1)
+	while ((ch = getopt(argc, argv, "+bcde:F:fhlMmNnoSstw")) != -1)
 		switch((char)ch) {
 		case 'b':
 			eflags |= VIS_NOSLASH;
diff --git a/src.freebsd/which/which.c b/src.freebsd/which/which.c
index dc22efb..2bad6b7 100644
--- a/src.freebsd/which/which.c
+++ b/src.freebsd/which/which.c
@@ -50,7 +50,7 @@ main(int argc, char **argv)
 
 	status = EXIT_SUCCESS;
 
-	while ((opt = getopt(argc, argv, "as")) != -1) {
+	while ((opt = getopt(argc, argv, "+as")) != -1) {
 		switch (opt) {
 		case 'a':
 			allpaths = 1;
@@ -141,4 +141,3 @@ print_matches(char *path, char *filename)
 	}
 	return (found ? 0 : -1);
 }
-
