diff --git a/src.freebsd/sed/main.c b/src.freebsd/sed/main.c
index 1f84249..36c96ca 100644
--- a/src.freebsd/sed/main.c
+++ b/src.freebsd/sed/main.c
@@ -133,7 +133,7 @@ main(int argc, char *argv[])
 	fflagstdin = 0;
 	inplace = NULL;
 
-	while ((c = getopt(argc, argv, "EI:ae:f:i:lnru")) != -1)
+	while ((c = getopt(argc, argv, "+EI:ae:f:i:lnru")) != -1)
 		switch (c) {
 		case 'r':		/* Gnu sed compat */
 		case 'E':
