diff --git a/src.freebsd/coreutils/expr/expr.y b/src.freebsd/coreutils/expr/expr.y
index 3dabad4..3e06f18 100644
--- a/src.freebsd/coreutils/expr/expr.y
+++ b/src.freebsd/coreutils/expr/expr.y
@@ -273,7 +273,7 @@ main(int argc, char *argv[])
 		av = argv + 1;
 		nonposix = 1;
 	} else {
-		while ((c = getopt(argc, argv, "e")) != -1) {
+		while ((c = getopt(argc, argv, "+e")) != -1) {
 			switch (c) {
 			case 'e':
 				nonposix = 1;
