--- a/crates/zed/resources/zed.desktop.in
+++ b/crates/zed/resources/zed.desktop.in
@@ -1,13 +1,13 @@
 [Desktop Entry]
 Version=1.0
 Type=Application
-Name=$APP_NAME
+Name=Zed
 GenericName=Text Editor
 Comment=A high-performance, multiplayer code editor.
-TryExec=$APP_CLI
-StartupNotify=$DO_STARTUP_NOTIFY
-Exec=$APP_CLI $APP_ARGS
-Icon=$APP_ICON
+TryExec=z
+StartupNotify=true
+Exec=z %U
+Icon=dev.zed.Zed
 Categories=Utility;TextEditor;Development;IDE;
 Keywords=zed;
 # To add Zed to "Open Folder With..." context menu, add `inode/directory` to the MimeType field (semicolon separated)
@@ -17,5 +17,5 @@
 Actions=NewWorkspace;
 
 [Desktop Action NewWorkspace]
-Exec=$APP_CLI --new $APP_ARGS
+Exec=z --new %U
 Name=Open a new workspace
