--- a/deps/LIEF/third-party/spdlog/include/spdlog/fmt/bundled/format.h	2026-05-03 11:17:31.440010459 +1000
+++ b/deps/LIEF/third-party/spdlog/include/spdlog/fmt/bundled/format.h	2026-05-03 11:17:26.269870901 +1000
@@ -44,6 +44,7 @@
 #  include <cmath>    // std::signbit
 #  include <cstddef>  // std::byte
 #  include <cstdint>  // uint32_t
+#  include <cstdlib>  // malloc/free
 #  include <cstring>  // std::memcpy
 #  include <limits>   // std::numeric_limits
 #  include <new>      // std::bad_alloc
