--- a/test/Makefile	2026-04-29 14:53:39.970286140 +1000
+++ b/test/Makefile	2026-04-29 14:53:50.710534654 +1000
@@ -1,6 +1,6 @@
-CC = gcc
+CC ?= gcc
 
-CFLAGS += -std=gnu99 -Wall -DFAKE_STAT -Werror -Wextra $(FAKETIME_COMPILE_CFLAGS) -U_FILE_OFFSET_BITS -U_TIME_BITS
+CFLAGS += -std=gnu99 -Wall -DFAKE_STAT -Wextra $(FAKETIME_COMPILE_CFLAGS) -U_FILE_OFFSET_BITS -U_TIME_BITS
 LDFLAGS += -lrt -lpthread
 
 SRC = timetest.c
